pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1049,6 +1069,39 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 		}
 	}
 
+	// TODO where to store harcoded strings
+	if env.Mesh.GetEnableAutoMtls().Value {
+		// Per envoy docs
+		// If an endpoint metadata's value under *envoy.transport_socket* does not match any
+		// *TransportSocketMatch*, socket configuration fallbacks to use the *tls_context* or
+		// *transport_socket* specified in this cluster
+		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
+			{
```

Comment:
can you declare the transport option as a global variable so that we dont have to allocate this over and over?